presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -511,12 +510,10 @@ public HiveRecordWriter(
             // todo adapt input types to the file types as Hive does
             for (int fileIndex = 0; fileIndex < fileColumnNames.size(); fileIndex++) {
                 String columnName = fileColumnNames.get(fileIndex);
-                Type fileColumnType = fileColumnTypes.get(fileIndex);
-                int inputIndex = inputColumnNames.indexOf(columnName);
-                Type inputType = inputColumnTypes.get(inputIndex);
+                HiveType fileColumnHiveType = fileColumnHiveTypes.get(fileIndex);
+                HiveType inputHiveType = inputColumnMap.get(columnName).getHiveType();
 
-                // HACK: solve coercion in generic way
-                if (!typesMatchForInsert(fileColumnType, inputType)) {
+                if (!fileColumnHiveType.equals(inputHiveType)) {
```

Comment:
I see. The original change was to facilitate inserting `VARCHAR(x)` into a Hive `String` column.

In order to adapt the `typesMatchForInsert` to work on Hive types instead of Presto types. I'll need to at least introduce the Hive `VARCHAR` type. Otherwise, the check won't be doing anything meaningful. I'll most likely drop the check and use `equals` for now.

The reason I changed this is that comparing the partition type and table type as Presto types are not correct. For example, 64-bit Hive int and 32-bit one will both map to BIGINT in Presto, and appear the same. So far Presto does not support type coercion between Hive table and Hive partition. It's on our plan. This change and your change (`typesMatchForInsert`), when it is implemented for Hive types, will be a good step toward supporting that.